### PR TITLE
Create VMs for UA Pipeline

### PR DESCRIPTION
**Pull request by julianpistorius**
Fixes https://github.com/az-digitalag/organization/issues/41

### CODE REVIEW

**Comment by julianpistorius** [comment]
@nathanhruby - Does this look OK?

**Comment by nathanhruby** [comment]
Hey this looks good, if this is a common pattern you may want to make a module for the network+bastion+security+user work so it can be easily reused. 

You should consider using a [backend](https://www.terraform.io/docs/backends/index.html) with remote state capability, possibly one that supports [workspaces](https://www.terraform.io/docs/state/workspaces.html), so your laptop isn't the central hub for infrastructure work.  This will enable teams to collaborate as well as house your statefile someplace safe.  Terraform Cloud offers this for free if you don't have a suitable backend available.

File: variables.tf [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+variable "privkey" {
+  default = "~/.ssh/id_rsa"
+}
+
+variable "postgresql_flavor" {
+  default = "medium1"
+}
+
+variable "postgresql_image" {
+  default = "ubuntu-16.04.raw"
```
Comment (robkooper) [inline variables.tf]: if possible I'd recommend going with 18.04 that will give you support till 2023, same for all of the images below.

File: 01_network.tf [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  name                = "${var.env_name}-router"
+  external_network_id = "${data.openstack_networking_network_v2.external.id}"
+}
+
+resource "openstack_networking_router_interface_v2" "router_interface_1" {
+  router_id = "${openstack_networking_router_v2.router_1.id}"
+  subnet_id = "${openstack_networking_subnet_v2.subnet_1.id}"
+}
+
+resource "openstack_networking_floatingip_v2" "postgresql_ip" {
```
Comment (robkooper) [inline 01_network.tf]: I don't think it is necessary to give postgresql a public IP.

**Comment by robkooper** [comment]
We use https://www.mongodb.com/products/ops-manager for backups for mongo (requires a VM with diskspace) as well as https://www.pgbarman.org/ for backing up postgres database.

File: 01_network.tf (hunk 2 above)
Comment (julianpistorius) [reply 01_network.tf]: Yes, you're right. I only had it there so I could temporarily use it as a bastion host to check on the MongoDB nodes, and then forgot to take it out. I'll remove this now.

File: variables.tf (hunk 1 above)
Comment (julianpistorius) [reply variables.tf]: Righto. Was going to ask you about that. Thanks.

Comment (julianpistorius) [reply variables.tf]: Addressed with: https://github.com/julianpistorius/terraref-terraform/commit/d63658dbb571bd66c2564ce8d9f7f91eae286dff

File: 01_network.tf (hunk 2 above)
Comment (julianpistorius) [reply 01_network.tf]: Addressed by: https://github.com/julianpistorius/terraref-terraform/commit/a2161f405021799cbbafaf8c97126729bdae3b9d

**Comment by julianpistorius** [comment]
> We use https://www.mongodb.com/products/ops-manager for backups for mongo (requires a VM with diskspace) as well as https://www.pgbarman.org/ for backing up postgres database.

Addressed:

- Ops Manager: https://github.com/julianpistorius/terraref-terraform/commit/ea1fa10de6aeca3b1e1ae35bc01dbcd20dd3ef4a
- Barman: https://github.com/julianpistorius/terraref-terraform/commit/13fdecb2458cd5c74bb3e5f76fb56b99113a244c

**Pull request closed without merging**